node/node.go

```diff
@@ -1394,10 +1367,57 @@ func (app *App) Start(ctx context.Context) error {
 			app.host.ID().String(), app.Config.Genesis.GenesisID().ShortString())
 	}
 
-	if err := app.startServices(ctx, appErr); err != nil {
+	if err := app.startServices(ctx); err != nil {
 		return err
 	}
 
+	// need post verifying service to start first
+	if preserve != nil {
```

Comment:
I feel like the whole block inside the if could be its own method `app.restore(preserve)`.

Also I think `chkptData` or similar could be a better name for `preserve`.

Reply:
but these are not checkpointData. at this point the checkpoint data is already saved in db.
here `preserve` means the data we want to preserve from the previous db, and is trying to save them to the new db via the gossip validation path.